### DISCUSSION TITLE
Unique features in this repo

### DISCUSSION THREAD

**Discussion by anddea** [anddea/revanced-patches#446] [Announcements]
### Unique features:

> [!NOTE]
> #### YouTube
> - Bypass image region restrictions: Option to change the alternative domain in the settings
> - Custom branding icon: New splash animations and new icons
> - Force player buttons background: Option to change the color and opacity
> - Force snackbar theme
> - Overlay buttons: Content descriptions for improved accessibility (for users with visual impairments)
> - Overlay buttons: Second downloader on long tap
> - Seekbar components: Set Cairo seekbar colors
> - Shorts: Hide disabled comment button
> - Shorts action buttons: Other icons
> - SponsorBlock: Draggable new segment window
> - Translations: Remove unselected languages from UI 
> - Visual preferences icons: Updated icons and added some new icons
> <details>
> <summary>Removed on refactor</summary>
> 
> - Hide player buttons background in settings
> </details>

> [!IMPORTANT]
> #### YT & YTM
> - Icons
>   - Vanced Light
>   - Vanced Dark
>   - XIsr Yellow
> 
> - "Don't show again" option for GMSCore battery optimization dialog
> 
> <details>
> <summary>Already implemented</summary>
>
> - YT Music: Visual preferences icons
> - YT - Settings: Show categories while searching preferences
> 
> - Icons:
>   - Custom branding (YT and YTM)
>     - AFN Blue
>     - AFN Red
>     - Other
>       - Icons refined
>       - Animation for MMT (others didn't provide animation files)
> 
>   - Player overlay buttons
>     - Rounded
>     - Thin (prev. Outline) - refined
>   - Shorts overlay buttons
>     - Outline - refined
>     - Outline circled
>     - TikTok
>   - Preferences
> 
> - Design fixes
>   
> - Custom wordmark header
> - Force hide chapters in player (implemented in settings)
> - Force hide fullscreen button in player (implemented in settings)
> - Hide live chat replay
> - Hide live chat messages
> - Hide Shorts live header
> - Hide videos based on duration
> - Hide videos based on views (with views more than specified value)
> - Overlay buttons
>   - Adjustable margin option (less buggy)
>   - Speed button (on long press second time it reverts back to default speed)
>   - Time ordered playlist from all channel videos
> - Premium header switch in settings
> - Search bar
> - Search bar UI 2.0
> - Settings patch
>   - Custom ReVanced Extended preference name
>   - Move RVX settings position in Settings
> - Start page (more options):
>   - "Watch later" playlist
>   - Less important
>     - Gaming
>     - Live
>     - Music
>     - Movies
>     - Sports
>     - Browse channels
> - Swipe controls (adjustable swipe area)
> - Tuck away preferences (implemented in settings)
> - Translations (remove unused languages and add custom language)
> - Whitelist
> - Shorts action buttons: Original Outline Icons
> - Default video quality: Option to disable the toast
> - Default video speed: Option to disable the toast
> </details>

**Comment by Aryan-Chauhan** [comment]
• Force hide fullscreen button in player UI.

• Force hide chapters in player UI.

• Hide redudant text in YT Settings page.

These three are now available in inotia00's RVX.

**Reply by anddea** [reply]
Thank you, I crossed out these options.

**Reply by Aryan-Chauhan** [reply]
Hmm.

• Hide videos based on views (with views more than specified value)

This is also now available in inotia00's RVX:
<details>
<summary>Screenshot</summary>

![Screenshot_2024-05-10-15-33-12-66_a9acfd2263b6b7a65bf2540b0d9449ca.jpg](https://github.com/anddea/revanced-patches/assets/91815107/f8fbeef4-8ab3-4279-868d-1c444163607e)

</details>

**Reply by anddea** [reply]
Although Force patches are different, just like for "Hide player buttons background", I have 2 options for it one is Force and second inside settings (this one also unique to this repo), but it's buggy for some people and slows down the player. Force patches are less buggy overall (except for Hide redudant text in YT Settings page, not possible to hide account and data saving with tuck away preferences).

**Reply by anddea** [reply]
> Hide videos based on views (with views more than specified value)
>
> This is also now available in inotia00's RVX

In inotia's repo it hides only videos with views less than specified value, that's why I wrote this option with details.

**Reply by anddea** [reply]
> Hide player buttons background

Added this one, "Change position of RVX in settings" and ~~"Premium header switch in settings"~~ to the list, if they are implemented, let me know.

**Comment by Aryan-Chauhan** [comment]
These are my feature requests on anddea's RVX repo that are implemented in inotia00's RVX:

#21

#22

#28

#29

#121

#251

**Reply by Mohammed1520** [reply]
https://github.com/anddea/revanced-patches/issues/23
Please can you add this into inotia repo, so he could add this feature to RVX

**Reply by Aryan-Chauhan** [reply]
> https://github.com/anddea/revanced-patches/issues/23
> Please can you add this into inotia repo, so he could add this feature to RVX

@anddea

**Reply by anddea** [reply]
Not sure what do you mean here? Could you elaborate? Keep fullscreen is keep landscape now and it works as it works, in the best possible way that was found for now.

**Reply by Aryan-Chauhan** [reply]
> Not sure what do you mean here? Could you elaborate? Keep fullscreen is keep landscape now and it works as it works, in the best possible way that was found for now. 

He is requesting to implement to #23 .

`Keep Landscape` doesn't work if you switch back & forth between apps.

**Reply by anddea** [reply]
Got it. This is an old issue, can't find a way to solve it.

**Comment by Francesco146** [comment]
custom header

**Reply by anddea** [reply]
What do you mean? It's not available in inotia's repo?

**Reply by Francesco146** [reply]
nope :(

**Reply by anddea** [reply]
Added `Custom wordmark header` to the list, thank you

**Reply by Morafoka69** [reply]
Please how do I use this feature @anddea? I've been meaning to use it but don't know how to. Do you have like a tutorial I can read or watch? I'm using RVX Manager amd Revancify, but use RVX Manager because of Dev releases.

Thank you in anticipation!

**Reply by anddea** [reply]
In the Manager, when you add this option, there's a gear icon in the card for `Custom branding wordmark header`, it says that you have to provide folder with all expected files in order for it to work. It's not just one image file, you have to create multiple images for each DPI folder. You can enable dev release in RVX Manager too.

**Comment by Francesco146** [comment]
the premium header switch is available:
![Screenshot_20240510_123921_RVX](https://github.com/anddea/revanced-patches/assets/49027005/3061607a-5eb8-422c-9bee-eb00519c78e0)

**Reply by anddea** [reply]
Crossed out, thank you.

**Comment by Ahmey1400** [comment]
Youtube music spoof certificate for android auto

**Reply by anddea** [reply]
Added, thank you

**Reply by Ahmey1400** [reply]
![Screenshot_20240510_143922_RVX Youtube](https://github.com/anddea/revanced-patches/assets/143447608/0eb7f25b-a25f-4414-9e99-b7316393b969)

Why these button logos wont show up

**Reply by anddea** [reply]
> Why these button logos wont show up

What do you mean? You can set different icons for RVX, in options.json or gear icon in `Visual preferences icons` patch in the Manager.

**Comment by s7san007** [comment]
Custom branding splash and welcome logo for YTM.

**Reply by anddea** [reply]
Added, thank you

**Comment by Ahmey1400** [comment]
Custom revanced settings name such as Advanced

**Reply by anddea** [reply]
It's already added as Settings patch > Custom ReVanced Extended preference name. Thank you.

**Reply by Ahmey1400** [reply]
I cant find it at patchs section

**Reply by Francesco146** [reply]
> I can't find it at patches section

you can change the custom name via patch-options. anddea meant that this feature that is unique in his repo is already added in the list

**Reply by Ahmey1400** [reply]
Ow ok sorry for my miss understood

**Comment by RioMeYeah** [comment]
Hello Anddea, I appreciate your work on these patches. Could you please continue to publish patches regularly? It seems 'Inotia' frequently stops and returns, and soon he may stop again. Your consistent efforts ensure access to these valuable updates. Also, does this patch "[v2.225.0-dev.10](https://github.com/anddea/revanced-patches/releases/tag/v2.225.0-dev.10)" mean you have all patches from "Inotia's repo," and now I can have everything from his last stable patch with this patch? Thanks.

**Reply by anddea** [reply]
> everything from his last stable patch

Yes, everything from his latest release. And additional features on top.

**Reply by RioMeYeah** [reply]
Thank you! When will your stable release be published?

**Reply by anddea** [reply]
> stable release

I wish I knew, but you can use dev, they are stable anyway and bugs can be in both dev and stable releases.

**Reply by RioMeYeah** [reply]
Thank you again, I wish you the best.

**Reply by AbakNacchan** [reply]
Unless inotia00 merged all of your features to his repo, I think we will stick to your repo with its bunch of useful features.

**Reply by GucekPL** [reply]
Question (because I'm curious),
Do you plan to cooperate with inotia in the future to bring your solutions to RVX (custom header, preference icons etc.) or do you prefer to create your own alternative to RVX?

**Reply by anddea** [reply]
> Do you plan to cooperate with inotia in the future to bring your solutions to RVX

I'd be glad to do it, but I'm having a hard time to contact him, be it in private or in public conversation. Currently, I opened one pull request in inotia's repository, no questions of mine were answered.

**Reply by AbakNacchan** [reply]
Kindly bring back those useful features that "removed" after code refactoring.

**Reply by MtndaleRedneck** [reply]
> > Do you plan to cooperate with inotia in the future to bring your solutions to RVX
> 
> I'd be glad to do it, but I'm having a hard time to contact him, be it in private or in public conversation. Currently, I opened one pull request in inotia's repository, no questions of mine were answered. 

🙏 Inotia does great work, but he can't be trusted not to leave again. It's annoying to flipflop between your two repositories. I use RVX Manager and have to switch the sources whenever I go from him to you. If the two of you could collaborate that would be awesome.

**Comment by vippium** [comment]
As you mentioned that`Hide player button background` is whitelisted. Isn't there anyway to use it rn.

**Reply by anddea** [reply]
You can use `Force hide player buttons background`, it won't have a switch option in settings though.

**Reply by vippium** [reply]
First I thought that I included it but it is a bug. Now I checked that I forgot to add it.😂😜

**Comment by Marat580** [comment]


**Reply by anddea** [reply]
This option is enabled by default, not sure why you have this issue. Go to Extended > Player > Disable player popup panels.

**Reply by Marat580** [reply]
You're right. Turned it on. I'm just confused by the new settings interface.
Thank you

**Comment by nintendork07** [comment]
edit: inotia00 has fixed this in revanced-patches-4.8.0-dev.6

thank you for your great work.

seems like the "hide action buttons" setting under player settings is broken in both inotia00's RVX & the anddea latest v2.225.0-dev.11 version.

buttons like remix, thanks & report still show even though the option to hide them specifically is on.

otherwise, everything seems to work really well.

**Comment by ChesterXXX** [comment]
@anddea Would it be possible to reimplement the whitelisting sponsorblock and whitelisting custom speed features in a later update? They were really useful!

**Reply by Francesco146** [reply]
working on it

**Comment by destroyer22719** [comment]
Hi there, what difference does the certificate spoof make for revanced extended music?

**Reply by anddea** [reply]
It should spoof for Android Auto if I remember correctly.

**Comment by 54m4d** [comment]
@anddea Hi, I would like to Tuck away preferences, but I was not able to find how, has this patch been removed? I checked the Settings but there are only 2 options, one to insert a position for RVX and the other to set a custom name for it, I tested with 225-dev.21.

**Reply by Aryan-Chauhan** [reply]
`General / Settings Menu / Here you go!`

**Reply by 54m4d** [reply]
> `General / Settings Menu / Here you go!`

Thanks!

**Comment by milan96129** [comment]


**Reply by anddea** [reply]
It is present there and even enabled by default, maybe you used different patches. If you use latest patches just open RVX settings and input "speed dialog" in the search bar.

**Reply by Aryan-Chauhan** [reply]
Kindly check settings properly, it's available for all users.

**Reply by milan96129** [reply]
I use Revanced manager v1.20.1. Can you please send me link for RVX manager ? In GitHub, I'm confused where to download it. or Is there any way I can apply that patch from Revanced Manager itself ?

**Reply by Aryan-Chauhan** [reply]
Here you go: https://github.com/inotia00/revanced-manager

Though, I like using ReVancify more: https://github.com/decipher3114/Revancify

**Reply by Aryan-Chauhan** [reply]
Settings / Player / Player Buttons / Show Speed Dialog Button.

**Reply by milan96129** [reply]
I was not aware about this until I found it on reddit :)
![revanced](https://github.com/anddea/revanced-patches/assets/142959279/6b66f985-d766-4ad9-baae-cd63a0f6e850)

**Comment by Kuddus73** [comment]
Hi @anddea,
Can you PR `Search bar for settings` to inotia?

**Reply by anddea** [reply]
There was [an issue](https://github.com/anddea/revanced-patches/issues?q=is%3Aissue+is%3Aclosed+search+settings+inotia+) already and [a question in the PR](https://github.com/anddea/revanced-patches/pulls?q=is%3Apr+is%3Aclosed+search+inotia) with this request.

**Comment by 54m4d** [comment]
@anddea I disabled this patch, but I can still swipe down to enter full screen, this feature is now available by default on YouTube, my question is if this patch can be removed in the next commit since it's already obsolete?

Thnx

<details>
<summary>Screenshot</summary>
<img src=https://github.com/anddea/revanced-patches/assets/52612348/4c073970-9ce3-4467-9865-c1d177148837>
</details>

**Reply by anddea** [reply]
Don't know what you're talking about as it works just fine.

**Reply by Francesco146** [reply]
i think some old spoof client broke that. seems fixed

**Reply by 54m4d** [reply]
> Don't know what you're talking about as it works just fine.

The patch doesn't do anything, the swipe to enter full screen just works regardless if the options is turned on or off. And what works just fine?

**Reply by anddea** [reply]
> And what works just fine?

Enable watch panel gestures. When I turn this option off it disables gestures, if I enable, gestures work.

**Reply by 54m4d** [reply]
This is strange, when I turn it off, the swipe gesture still works, that's why I thought it was enabled by default in stock YT.
I don't know what else could affect this though, I'm using 19.20.34

**Reply by anddea** [reply]
> 19.20.34

Same. And it works for me as it should be. Maybe try to reinstall, or maybe some other setting interfering, or maybe it's some A/B.

**Reply by 54m4d** [reply]
I found out it's the latter.

**Comment by ntnguyen1234** [comment]
Hi. Are the audio / video codec patches (force VP9) no longer in version 2.225? I applied this patch to YouTube 19.20.34 and the number of patches that appear are much lower than previous one (54 vs > 100)

![Capture+_2024-05-29-20-01-51.png](https://github.com/anddea/revanced-patches/assets/23257168/b721b283-b907-44de-9ab2-cc710506c1b6)

Thank you.

**Reply by Aryan-Chauhan** [reply]
Many different patches are merged into single patches in latest updates.

`Force VP9` doesn't exist anymore, but these two are added by inotia00:

![Screenshot_2024-05-29-18-56-51-81_fe557b5c831afa8eecca413bd340ebe7.jpg](https://github.com/anddea/revanced-patches/assets/91815107/c588a9d9-a916-46e0-aecf-dbb83044f1d2)

**Reply by ntnguyen1234** [reply]
Thanks. I enabled the above 2 toggles, but the highest quality option is still 720p60

<details>

![Capture+_2024-05-29-21-08-57.png](https://github.com/anddea/revanced-patches/assets/23257168/969f7267-ddf3-4d95-8bc2-ee871783a795)

</details>

**Reply by Aryan-Chauhan** [reply]
> Thanks. I enabled the above 2 toggles, but the highest quality option is still 720p60
> 
> <details>
> 
> ![Capture+_2024-05-29-21-08-57.png](https://github.com/anddea/revanced-patches/assets/23257168/969f7267-ddf3-4d95-8bc2-ee871783a795)
> 
> </details>

You can turn on `Spoof Device Dimensions`, then clear cache of RVX app, then reboot your phone.

**Comment by Aryan-Chauhan** [comment]
Hi.

After the PR merge, many new patches/options are available in inotia00's RVX.

Can you update this discussion thread?

**Reply by anddea** [reply]
Can you do it for me? Just give me the list of unique features, including new. As I know, he didn't merge everything.

**Reply by Aryan-Chauhan** [reply]
Sure, both RVX are already installed on my phone.

Below features are now available in inotia00 patches, so you can strikethrough them in the main list.

```
For YT:

• AFN Blue icon

• AFN Red icon

• MMT Splash Animation

• Outline Icons in Shorts Overlay Buttons.

• Outline Circled Icons in Shorts Overlay Buttons.

• Custom Wordmark Header

• Hide Shorts Live Header

• Search Bar

• Custom RVX Settings name

• Move RVX Settings position in Settings

• Translations (remove unused languages and add custom language)
```

```
For YT Music:

• Custom Banding Icon (splash and welcome icons)
```

**Reply by s7san007** [reply]
+ • TikTok (Round) Icons in Shorts Overlay Buttons.

**Reply by anddea** [reply]
Updated, let me know if something's missing

**Comment by Xisrr1** [comment]
Add the new splash animation

**Reply by anddea** [reply]
Added

**Comment by Aryan-Chauhan** [comment]
New search bar is now available on inotia00 patches as well.

**Reply by Francesco146** [reply]
thanks

**Comment by Aryan-Chauhan** [comment]
`Shorts Action Buttons / Original Outline Icons` are now available in inotia00 patches also.

**Reply by Francesco146** [reply]
dould you please check if the post is up to date, when you have the will and time?

**Reply by Aryan-Chauhan** [reply]
I just checked and `Bypass Image Region Restrictions` is available in inotia00 patches.

**Reply by anddea** [reply]
Is there an option to change alternative domain? I believe it's not there, there's only toggle, but no setting to provide your own domain.

**Comment by 54m4d** [comment]
I see that patches from here are added to Inotia's, what is the end goal? And which repo is more recommended?

**Reply by Francesco146** [reply]
use the one you like more, if your configuration is based on the features listed above then use this repo, otherwise use RVX

**Reply by anddea** [reply]
There are some differences though. Like settings structure / order. Or some patches are different, like custom branding icon here includes header, while in inotia's is a separate patch.

**Reply by 54m4d** [reply]
Thanks

**Comment by Aryan-Chauhan** [comment]
`Don't show again` option for GMSCore battery optimization dialog.

This is added now in inotia00 dev patches.

**Reply by anddea** [reply]
I don't see it in his repo, what do you mean?

**Reply by Aryan-Chauhan** [reply]
> I don't see it in his repo, what do you mean?

You have to set this patch option as `false`.

![Screenshot_2024-07-18-14-57-27-68_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/user-attachments/assets/ba92a762-fc9e-4a0e-afb3-d6a33a24b0e3)

**Reply by anddea** [reply]
It's a little different thing. It checks whether GmsCore is installed on the device or not, it won't show any warnings even if GmsCore is not installed, so it's recommended to keep it true anyway. Don't show again is only for battery optimization (since some people see it even if all battery optimizations are disabled), and there's a toggle in the settings, it doesn't work for don't kill my app though (it can be implemented but YT/YTM won't work if GmsCore is killed but can work without battery optimizations).

**Reply by Aryan-Chauhan** [reply]
Well, it serves the same functionality for me.

**Comment by Xisrr1** [comment]
`Visual preferences icons` is now in RVX. 

`Revancify Yellow` to `Xisr Yellow`.

**Comment by Morafoka69** [comment]
Hello.
Please I noticed that YT Dislike is not showing, not even the likes. I'm using the Latest Anddea Patches and the latest recommended YT version. Please what do you think can be done to fix this?
Thanks in anticipation. 
![Screenshot_20241009-025531](https://github.com/user-attachments/assets/3b0ed33d-d2e0-4b46-9845-5ec3522883a0)

**Reply by anddea** [reply]
The issue cannot be reproduced on my side. The only solution that comes up is to reinstall the app.

**Reply by Morafoka69** [reply]
This fixed it. Thanks a bunch. 


Please I'm having an issue with YT Music. There usually used to be a button at the top corner of the player instead of the cast button that changes video to the main audio and it's no longer there. How do I bring it back? I'm using the Latest Dev patches.

Thank you in anticipation.

**Reply by anddea** [reply]
There's an option Hide Audio & Video toggle is RVX > Player. But it should be turned off by default. So you can just reset your default settings.

**Comment by Morafoka69** [comment]
You don't understand. I want that button because whenever I play a song on YT Music and it starts playing the music video or lyric video version, that button switches it to the original album version.

Please just watch this short vid if you don't understand what I'm saying:

https://imgur.com/a/replace-cast-button-O1cL4lb

Thank you in anticipation.

**Reply by anddea** [reply]
I've never seen this button. How did you get it before?

**Comment by Morafoka69** [comment]
Thanks for replying. 

I was using inotia's patches and later switched to yours when you started. It has to have been in yours because your patches are forked from inotia's, right?

This is the link of inotia's patches with screenshots:

https://github.com/ReVanced-Extended-Community/Patches-Documentation/tree/main?tab=readme-ov-file#youtube-music-1

This is the name of the patch:
![Screenshot_20241030-132339](https://github.com/user-attachments/assets/7447864e-6d93-4a40-aa4b-088295a0c4ad)

Thanks in advance.

**Reply by anddea** [reply]
I see. It was removed in June, read this comment https://github.com/inotia00/ReVanced_Extended/issues/2101#issuecomment-2140459572.

Here's the release when it was removed [inotia00/v4.9.1-dev.1](https://github.com/inotia00/revanced-patches/releases/tag/v4.9.1-dev.1) and [anddea/v2.226.0-dev.6](https://github.com/anddea/revanced-patches/releases/tag/v2.226.0-dev.6).

**Reply by Charu-Mahbub** [reply]
How to use my own YouTube Icon instead of using Default Icons in the Options. How I will locate icon path in options.json file? btw I like to use Revanced-CLI  for using your patch. @anddea

**Reply by anddea** [reply]
> How to use my own YouTube Icon instead of using Default Icons in the Options. How I will locate icon path in options.json file? btw I like to use Revanced-CLI for using your patch. @anddea

Just provide the path to the folder containing the icons, you can check this video tutorial: https://t.me/AnddeaChat/15661. For assistance, you can ask in the Telegram chats: https://t.me/AnddeaChat or https://t.me/revanced_extended_chat.

**Reply by Charu-Mahbub** [reply]
Thanks It Worked. But When giving full path in "revanced-cli" was patching failed. It worked when I just gave icon folder name in the option.json file. May be full path is necessary for revancify on android. Btw, If poossible create patch for Facebook and Facebook Messenger also. Thanks in Advance.
